src/jsoncharutils_tables.h

```diff
@@ -215,652 +215,663 @@ static const double power_of_ten[] = {
-    0x88aaa1845b8fdd0, 0x8aad549e57273d45,
-    0x36ac54e2f678864b, 0x84576a1bb416a7dd,
-    0x656d44a2a11c51d5, 0x9f644ae5a4b1b325,
-    0x873d5d9f0dde1fee, 0xa90cb506d155a7ea,
-    0x9a7f12442d588f2, 0xc11ed6d538aeb2f,
-    0x8f1668c8a86da5fa, 0xf96e017d694487bc,
-    0x37c981dcc395a9ac, 0x85bbe253f47b1417,
-    0x93956d7478ccec8e, 0x387ac8d1970027b2,
-    0x6997b05fcc0319e, 0x441fece3bdf81f03,
-    0xd527e81cad7626c3, 0x8a71e223d8d3b074,
-    0xf6872d5667844e49, 0xb428f8ac016561db,
-    0xe13336d701beba52, 0xecc0024661173473,
-    0x27f002d7f95d0190, 0x31ec038df7b441f4,
-    0x7e67047175a15271, 0xf0062c6e984d386,
-    0x52c07b78a3e60868, 0xa7709a56ccdf8a82,
-    0x88a66076400bb691, 0x6acff893d00ea435,
-    0x583f6b8c4124d43, 0xc3727a337a8b704a,
-    0x744f18c0592e4c5c, 0x1162def06f79df73,
-    0x8addcb5645ac2ba8, 0x6d953e2bd7173692,
-    0xc8fa8db6ccdd0437, 0x1d9c9892400a22a2,
-    0x2503beb6d00cab4b, 0x2e44ae64840fd61d,
-    0x5ceaecfed289e5d2, 0x7425a83e872c5f47,
-    0xd12f124e28f77719, 0x82bd6b70d99aaa6f,
-    0x636cc64d1001550b, 0x3c47f7e05401aa4e,
-    0x65acfaec34810a71, 0x7f1839a741a14d0d,
-    0x1ede48111209a050, 0x934aed0aab460432,
-    0xf81da84d5617853f, 0x36251260ab9d668e,
-    0xc1d72b7c6b426019, 0xb24cf65b8612f81f,
-    0xdee033f26797b627, 0x169840ef017da3b1,
-    0x8e1f289560ee864e, 0xf1a6f2bab92a27e2,
-    0xae10af696774b1db, 0xacca6da1e0a8ef29,
-    0x17fd090a58d32af3, 0xddfc4b4cef07f5b0,
-    0x4abdaf101564f98e, 0x9d6d1ad41abe37f1,
-    0x84c86189216dc5ed, 0x32fd3cf5b4e49bb4,
-    0x3fbc8c33221dc2a1, 0xfabaf3feaa5334a,
-    0x29cb4d87f2a7400e, 0x743e20e9ef511012,
-    0x914da9246b255416, 0x1ad089b6c2f7548e,
-    0xa184ac2473b529b1, 0xc9e5d72d90a2741e,
-    0x7e2fa67c7a658892, 0xddbb901b98feeab7,
-    0x552a74227f3ea565, 0xd53a88958f87275f,
-    0x8a892abaf368f137, 0x2d2b7569b0432d85,
-    0x9c3b29620e29fc73, 0x8349f3ba91b47b8f,
-    0x241c70a936219a73, 0xed238cd383aa0110,
-    0xf4363804324a40aa, 0xb143c6053edcd0d5,
-    0xdd94b7868e94050a, 0xca7cf2b4191c8326,
-    0xfd1c2f611f63a3f0, 0xbc633b39673c8cec,
-    0xd5be0503e085d813, 0x4b2d8644d8a74e18,
-    0xddf8e7d60ed1219e, 0xcabb90e5c942b503,
-    0x3d6a751f3b936243, 0xcc512670a783ad4,
-    0x27fb2b80668b24c5, 0xb1f9f660802dedf6,
-    0x5e7873f8a0396973, 0xdb0b487b6423e1e8,
-    0x91ce1a9a3d2cda62, 0x7641a140cc7810fb,
-    0xa9e904c87fcb0a9d, 0x546345fa9fbdcd44,
-    0xa97c177947ad4095, 0x49ed8eabcccc485d,
-    0x5c68f256bfff5a74, 0x73832eec6fff3111,
-    0xc831fd53c5ff7eab, 0xba3e7ca8b77f5e55,
-    0x28ce1bd2e55f35eb, 0x7980d163cf5b81b3,
-    0xd7e105bcc332621f, 0x8dd9472bf3fefaa7,
-    0xb14f98f6f0feb951, 0x6ed1bf9a569f33d3,
-    0xa862f80ec4700c8, 0xcd27bb612758c0fa,
-    0x8038d51cb897789c, 0xe0470a63e6bd56c3,
-    0x1858ccfce06cac74, 0xf37801e0c43ebc8,
-    0xd30560258f54e6ba, 0x47c6b82ef32a2069,
-    0x4cdc331d57fa5441, 0xe0133fe4adf8e952,
-    0x58180fddd97723a6, 0x570f09eaa7ea7648,};
-
+const uint64_t power_of_five_128[]= {
```

Comment:
My reading must have been sloppy somehow. Given that we switched from 10^x to 5^x, where in the code do we add the other 2^x back in?

Reply:
The 2^x is not part of the binary significand.

We are mapping w * 10 ^q to m * 2^p but w * 10 ^q = w * 5^q * 2^q. The 2^q is matched by the  2^p.

I will add a remark in the code. Something like this:


/**
 * When mapping numbers from decimal to binary,
 * we go from w * 10^q to m * 2^p but we have
 * 10^q = 5^q * 2^q, so effectively
 * we are trying to match
 * w * 2^q * 5^q to m * 2^p. Thus the powers of two
 * are not a concern since they can be represented
 * exactly using the binary notation, only the powers of five
 * affect the binary significand.
 */